### PR TITLE
[expo] Upgrade @react-native-community/datetimepicker to `8.2.0`

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-13680

# How

Upgrades  @react-native-community/datetimepicker to `8.2.0`

# Test Plan

- expo go  
- bare-expo 

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by alanjhughes — APPROVED** [review]

File: apps/bare-expo/ios/Podfile.lock [hunk 1]
```diff
@@ -3186,8 +3186,8 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   BenchmarkingModule: 0be4117c4bf255d2e21df9e4a4a486ec89c8640f
-  boost: 4cb898d0bf20404aab1850c656dcea009429d6c1
-  DoubleConversion: 76ab83afb40bddeeee456813d9c04f67f78771b5
+  boost: 1dca942403ed9342f98334bf4c3621f011aa7946
+  DoubleConversion: f16ae600a246532c4020132d54af21d0ddb2a385
```
Comment (Kudo) [inline apps/bare-expo/ios/Podfile.lock:3190]: also unrelated hash changes

File: apps/native-component-list/package.json [hunk 2]
```diff
@@ -36,8 +36,8 @@
     "@expo/react-native-action-sheet": "^3.8.0",
     "@expo/styleguide-base": "^1.0.1",
     "@react-native-async-storage/async-storage": "1.23.1",
-    "@react-native-community/datetimepicker": "8.0.0",
-    "@react-native-community/netinfo": "11.3.1",
+    "@react-native-community/datetimepicker": "8.2.0",
+    "@react-native-community/netinfo": "11.4.1",
```
Comment (Kudo) [inline apps/native-component-list/package.json:40]: is netinfo bump intentional in this pr?

File: apps/native-component-list/package.json [hunk 3]
```diff
@@ -137,8 +137,8 @@
     "react-native": "0.75.2",
     "react-native-dropdown-picker": "^5.3.0",
     "react-native-gesture-handler": "~2.20.0",
-    "react-native-maps": "1.14.0",
-    "react-native-pager-view": "6.4.1",
+    "react-native-maps": "1.18.0",
+    "react-native-pager-view": "6.4.0",
```
Comment (Kudo) [inline apps/native-component-list/package.json:141]: unrelated maps, pager view changes

**Review by Kudo — APPROVED** [review]

File: apps/native-component-list/package.json (hunk 2 above)
Comment (gabrieldonadel) [reply apps/native-component-list/package.json:40]: Sorry, for this 🤦 something seems wrong with my git

File: apps/bare-expo/ios/Podfile.lock (hunk 1 above)
Comment (gabrieldonadel) [reply apps/bare-expo/ios/Podfile.lock]: FIxed

File: apps/native-component-list/package.json (hunk 3 above)
Comment (gabrieldonadel) [reply apps/native-component-list/package.json]: fixed
